pmlc/ast/builder.cc

```diff
@@ -720,10 +720,14 @@ struct ProgramBuilder {
       }
       assert(op && "Unexpected block argument");
       Operation *clonedOp = bodyBuilder.clone(*op, mapper);
-      if (results.contains(value)) {
-        for (Value result : clonedOp->getResults()) {
-          innerResults.push_back(result);
+
+      auto iter = std::find(results.begin(), results.end(), value);
+      if (iter != results.end()) {
+        if (clonedOp->getNumResults() != 1) {
+          throw std::runtime_error("Program build failure.");
```

Comment:
This error message will be confusing to users.

Reply:
Thanks. I updated this section of code with support for ops that has multiple results and avoided throwing an error here.